New compiler warning on Windows

From https://cran.rstudio.org/web/checks/check_results_s2.html / https://www.r-project.org/nosvn/R.check/r-devel-windows-x86_64/s2-00check.html

```
Result: WARN
    Found the following significant warnings:
     ../src/s2/s2shape_index.h:276:21: warning: 'template<class _Category, class _Tp, class _Distance, class _Pointer, class _Reference> struct std::iterator' is deprecated [-Wdeprecated-declarations]
     ../src/s2/s2builder_graph.h:182:21: warning: 'template<class _Category, class _Tp, class _Distance, class _Pointer, class _Reference> struct std::iterator' is deprecated [-Wdeprecated-declarations]
    See 'd:/Rcompile/CRANpkg/local/4.3/s2.Rcheck/00install.out' for details.
```

--- a/src/Makevars.in
+++ b/src/Makevars.in
@@ -1,7 +1,6 @@
 PKG_CPPFLAGS = -I../src -DSTRICT_R_HEADERS
 PKG_LIBS = @libs@
 PKG_CXXFLAGS = @cflags@ -pthread
-CXX_STD = CXX11
 
 ABSL_LIBS = absl/base/internal/cycleclock.o \
     absl/base/internal/low_level_alloc.o \

--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -2,8 +2,6 @@ VERSION = 1.1.1k
 PKG_CPPFLAGS = -DS2_USE_EXACTFLOAT -D_USE_MATH_DEFINES -DNDEBUG -DIS_LITTLE_ENDIAN -DOMIT_STRPTIME -I../windows/openssl-$(VERSION)/include -I../src
 PKG_LIBS = -Ls2 -ls2static -L../windows/openssl-$(VERSION)/lib${R_ARCH}${CRT} -lssl -lcrypto -lws2_32 -lgdi32 -lcrypt32
 
-CXX_STD = CXX11
-
 STATLIB = s2/libs2static.a
 
 ABSL_LIBS = absl/base/internal/cycleclock.o \

